pkg/services/tree/service.go

```diff
@@ -48,18 +51,25 @@ func New(opts ...Option) *Service {
 	s.replicateCh = make(chan movePair, s.replicatorChannelCapacity)
 	s.replicationTasks = make(chan replicationTask, s.replicatorWorkerCount)
 	s.containerCache.init(s.containerCacheSize)
+	s.cnrMap = make(map[cidSDK.ID]struct{})
+	s.syncChan = make(chan struct{})
 
 	return &s
 }
 
 // Start starts the service.
 func (s *Service) Start(ctx context.Context) {
 	go s.replicateLoop(ctx)
+	go s.syncLoop(ctx)
+
+	// initial sync
+	s.syncChan <- struct{}{}
 }
 
 // Shutdown shutdowns the service.
 func (s *Service) Shutdown() {
 	close(s.closeCh)
+	close(s.syncChan)
```

Comment:
Can we (1) drain it here first and then check `<-s.closeCh` in syncAll before blocking. This way we avoid possible panic.